metrics/src/test/java/tech/pegasys/pantheon/metrics/prometheus/PrometheusMetricsSystemTest.java

```diff
@@ -0,0 +1,137 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.metrics.prometheus;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.pantheon.metrics.MetricCategory.JVM;
+import static tech.pegasys.pantheon.metrics.MetricCategory.PEERS;
+import static tech.pegasys.pantheon.metrics.MetricCategory.RPC;
+
+import tech.pegasys.pantheon.metrics.Counter;
+import tech.pegasys.pantheon.metrics.LabelledMetric;
+import tech.pegasys.pantheon.metrics.MetricsSystem;
+import tech.pegasys.pantheon.metrics.Observation;
+import tech.pegasys.pantheon.metrics.OperationTimer;
+import tech.pegasys.pantheon.metrics.OperationTimer.TimingContext;
+
+import java.util.Comparator;
+
+import org.junit.Test;
+
+public class PrometheusMetricsSystemTest {
+
+  private static final Comparator<Observation> IGNORE_VALUES =
+      Comparator.comparing(Observation::getCategory)
+          .thenComparing(Observation::getMetricName)
+          .thenComparing((o1, o2) -> o1.getLabels().equals(o2.getLabels()) ? 0 : 1);
+
+  private final MetricsSystem metricsSystem = new PrometheusMetricsSystem();
+
+  @Test
+  public void shouldCreateObservationFromCounter() {
+    final Counter counter = metricsSystem.createCounter(PEERS, "connected", "Some help string");
+
+    counter.inc();
+    assertThat(metricsSystem.getMetrics())
+        .containsExactly(new Observation(PEERS, "connected", 1d, emptyList()));
+
+    counter.inc();
+    assertThat(metricsSystem.getMetrics())
+        .containsExactly(new Observation(PEERS, "connected", 2d, emptyList()));
+  }
+
+  @Test
+  public void shouldCreateSeparateObservationsForEachCounterLabelValue() {
+    final LabelledMetric<Counter> counter =
+        metricsSystem.createCounter(PEERS, "connected", "Some help string", "labelName");
+
+    counter.labels("value1").inc();
+    counter.labels("value2").inc();
+    counter.labels("value1").inc();
+
+    assertThat(metricsSystem.getMetrics())
+        .containsExactlyInAnyOrder(
+            new Observation(PEERS, "connected", 2d, singletonList("value1")),
+            new Observation(PEERS, "connected", 1d, singletonList("value2")));
+  }
+
+  @Test
+  public void shouldCreateObservationsFromTimer() {
+    final OperationTimer timer = metricsSystem.createTimer(RPC, "request", "Some help");
+
+    final TimingContext context = timer.startTimer();
```

Comment:
Should we make one of these timer tests a try with resources?  That would validate the Closable interface on them.

Reply:
Done.